contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -629,11 +636,135 @@ contract UniV3StablesJoint is NoHedgeJoint {
     function _positionInfo()
         private
         view
-        returns (IUniswapV3Pool.PositionInfo memory)
+        returns (
+            uint128 liquidity,
+            uint256 feeGrowthInside0LastX128,
+            uint256 feeGrowthInside1LastX128,
+            uint128 tokensOwed0,
+            uint128 tokensOwed1
+        )
     {
         bytes32 key = keccak256(
             abi.encodePacked(address(this), minTick, maxTick)
         );
         return IUniswapV3Pool(pool).positions(key);
     }
+
+    /*
+     * @notice
+     *  Function used by governance to swap tokens manually if needed, can be used when closing 
+     * the LP position manually and need some re-balancing before sending funds back to the 
+     * providers
+     * @param swapPath, path of addresses to swap, should be 2 and always tokenA <> tokenB
+     * @param swapInAmount, amount of swapPath[0] to swap for swapPath[1]
+     * @param minOutAmount, minimum amount of want out
+     * @return swapped amount
+     */
+    function swapTokenForTokenManually(
+        bool sellA,
+        uint256 swapInAmount,
+        uint256 
+    ) external onlyGovernance override returns (uint256) {
+
+        if(sellA) {
+            return swap(
+                tokenA,
+                tokenB,
+                swapInAmount
+                );
+        } else {
+            return swap(
+                tokenB,
+                tokenA,
+                swapInAmount
+                );
+        }
+
+        
+    }
+
+    /*
+     * @notice
+     *  Function used by keepers to assess whether to harvest the joint and compound generated
+     * fees into the existing position, checks whether both pending fee amounts (tokenA and B)
+     * are greater than minRewardToHarvest
+     * @param callCost, call cost parameter
+     * @return bool amount assessing whether to harvest or not
+     */
+    function harvestTrigger(uint256 callCost) external view override returns (bool) {
+
+        uint256 _minRewardToHarvest = minRewardToHarvest;
+        if (_minRewardToHarvest == 0) {
+            return false;
+        }
+
+        uint256[] memory _pendingRewards = pendingRewards();
+
+        if (_pendingRewards[0] >= _minRewardToHarvest * (10**IERC20Extended(rewardTokens[0]).decimals()) / RATIO_PRECISION && 
```

Comment:
I think I would replace && with || here but I see it would be very rare to see to have rewards only on one side